lib/moku/task/symlink.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+require "moku/task/task"
+require "moku/sites/scope"
+
+module Moku
+  module Task
+
+    # Create a symlink for eac path pair defined in infrastructure.yml
+    class Symlink < Task
+
+      def call(release)
+        release.run(Sites::Scope.all, command)
+      end
+
+      private
+
+      def command
+        <<~'CMD'
+          ruby -e 'require "yaml"; YAML.load_file("infrastructure.yml")["path"].each{|k,v| `ln -s #{v} #{k}`} if File.exist?("infrastructure.yml")'
```

Comment:
Will this infrastructure.yaml file not be present on the deploy host? It seems like it would be cleaner to load that yaml and construct the list of symlinks, then just run ln -s commands remotely. Maybe I'm missing something, though.

Reply:
Currently, for multisite support, the artifact gets built with a `sites` folder that might have some files like `sites/macc/infrastructure.yml` and another `sites/hatcher/infrastructure.yml`, which differ. These are overlaid on the correct hosts during the release phase, in `Task::OverlaySites`. 

I think what you suggest would work, but the task would have to know about that `sites` folder's makeup, construct the commands, then forward them along to the release. I think it ends up being a lot of complexity for something that is still less ideal than constructing the site-specific artifacts earlier in the process. That's something I want to work on, but this commit fulfills the functionality without locking us into some strategy; i.e., I think this is the best stopgap.

Reply:
OK, makes sense.